File: 프로그래머스/lv1/12928. 약수의 합/약수의 합.py
def solution(n):
    mid = int(n ** (1/2))
    divisor = []
    for i in range(1, mid + 1):
        if n % i == 0:
            divisor.append(i)
            divisor.append(n/i)
            
    
    answer = sum(divisor)
    return answer - mid if (n ** (1/2)).is_integer() else answer

File: 프로그래머스/lv1/12928. 약수의 합/약수의 합.java
class Solution {
    public int solution(int n) {
        int answer = 0;
        double sqrt = Math.sqrt(n);
        if (sqrt == (int) sqrt) {
            answer -= (int) sqrt;
        }

        for (int i = 1; i <= (int)sqrt; i++) {
            if (n % i == 0) {
                answer += i + (n / i);
            }
        }
        return answer;
    }
}